### PR TITLE
Ibta 53 create employee

### PR DESCRIPTION
**Pull request by GytisMas**

### CODE REVIEW

File: src/CSI.IBTA.Employer/Controllers/EmployeesController.cs [hunk 1]
```diff
@@ -64,5 +65,53 @@ public EmployeesController(IEmployeesClient employeeClient)
 
             return PartialView("_Employees", viewModel);
         }
+
+        [HttpGet("/CreateEmployee")]
```
Comment (radislavkireev) [inline src/CSI.IBTA.Employer/Controllers/EmployeesController.cs]:
```suggestion
        [HttpGet("CreateEmployee")]
```

File: src/CSI.IBTA.Employer/Controllers/EmployeesController.cs [hunk 2]
```diff
@@ -64,5 +65,53 @@ public EmployeesController(IEmployeesClient employeeClient)
...
+            var viewModel = new EmployeeViewModel()
+            {
+                ActionName = "CreateEmployee",
+                EmployerId = employerId,
+                DateOfBirth = DateOnly.FromDateTime(DateTime.UtcNow)
+            };
+            return PartialView("_EmployeeForm", viewModel);
+        }
+
+        [HttpPost("/CreateEmployee")]
```
Comment (radislavkireev) [inline src/CSI.IBTA.Employer/Controllers/EmployeesController.cs]:
```suggestion
        [HttpPost("CreateEmployee")]
```

File: src/CSI.IBTA.UserService/Services/EmployeesService.cs [hunk 3]
```diff
@@ -42,5 +45,67 @@ public async Task<GenericResponse<PagedEmployeesResponse>> GetEmployees(int page
...
+                return new GenericResponse<EmployeeDto>(new HttpError("An employee already exists with the same name.", HttpStatusCode.BadRequest), null);
+            }
+
+            bool hasSameUsername = await _userUnitOfWork.Users.GetSet().AnyAsync(x => x.Account.Username == dto.UserName);
+            if (hasSameUsername)
+            {
+                return new GenericResponse<EmployeeDto>(new HttpError("An employee already exists with the same username.", HttpStatusCode.BadRequest), null);
+            }
+
+            var e = new User()
```
Comment (radislavkireev) [inline src/CSI.IBTA.UserService/Services/EmployeesService.cs]:
```suggestion
            var user = new User()
```
lets have semantic namings

File: src/CSI.IBTA.UserService/Mapping/MapperProfile.cs [hunk 4]
```diff
@@ -17,8 +17,8 @@ public MapperProfile()
                     user.Lastname,
                     user.AccountId,
                     user.Employer != null ? user.Employer.Id : null,
-                    user.Emails[0].EmailAddress,
-                    user.Phones[0].PhoneNumber)
+                    user.Emails != null && user.Emails.Count > 0 ? user.Emails[0].EmailAddress : null,
+                    user.Phones != null && user.Phones.Count > 0 ? user.Phones[0].PhoneNumber : null)
```
Comment (VidasBui) [inline src/CSI.IBTA.UserService/Mapping/MapperProfile.cs]: EmailAdress and PhoneNumber are not nullable for UserDto, so if employer does not have one, we should assign an empty string. 
```suggestion
                    user.Phones != null && user.Phones.Count > 0 ? user.Phones[0].PhoneNumber : "")
```

This prevents us from dealing with null strings, as attempting to perform operations on a null string (such as accessing its properties or calling methods) can lead to a `NullReferenceException`

**Review by VidasBui — APPROVED** [review]

**Review by radislavkireev — APPROVED** [review]

**Pull request merged**